app/customer.py

```diff
@@ -0,0 +1,54 @@
+import datetime
+from app.car import Car
+from app.shop import Shop
+
+
+class Customer:
+    def __init__(self, name: str,
+                 product_cart: dict,
+                 location: list,
+                 money: int | float,
+                 car: Car) -> None:
+        self.name = name
+        self.product_cart = product_cart
+        self.location = location
+        self.money = money
+        self.car = car
+
+    def calculate_trip_cost(self, shop: Shop) -> float:
+        distance_to_shop = self.calculate_distance(shop.location)
+        fuel_cost_to_shop = self.car.calculate_fuel_cost(distance_to_shop)
+        product_cost = self.calculate_product_cost(shop)
+        fuel_cost_home = self.car.calculate_fuel_cost(distance_to_shop)
+        return fuel_cost_to_shop + product_cost + fuel_cost_home
+
+    def calculate_distance(self, location: list) -> float:
+        return ((self.location[0] - location[0]) ** 2
+                + (self.location[1] - location[1]) ** 2) ** 0.5
+
+    def calculate_product_cost(self, shop: Shop) -> float:
+        cost = 0
+        for product, quantity in self.product_cart.items():
+            cost += shop.products[product] * quantity
+        return cost
```

Comment:
This is a code quality issue. You are calculating the product cost twice: once in the `calculate_product_cost` method and once in the `buy_products` method. It would be more efficient to calculate it once and store the result in a variable that can be used in both methods.